test/testcne/metrics_test.c

```diff
@@ -0,0 +1,101 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2022 Intel Corporation
+ */
+
+#include <stdio.h>           // for NULL, EOF
+#include <metrics.h>         // for metrics_register
+#include <tst_info.h>        // for tst_error, tst_end, tst_start, TST_FAILED
+
+#include "metrics_test.h"
+
+static int
+test_cb(metrics_client_t *client, const char *cmd __cne_unused, const char *params __cne_unused)
+{
+    cne_printf("Test Callback\n");
+    metrics_append(client, "\"name\":\"Metrics Test\"");
+    return 0;
+}
+
+static int
+metrics_test(void)
+{
+    const char *c;
+    int ret = 0;
+    metrics_client_t *client;
+    lport_stats_t stats = {0};
+    char lpname[]       = "test_lport";
+
+    client = calloc(1, sizeof(metrics_client_t));
+    if (client != NULL) {
+        client->cmd    = "/test";
+        client->params = "params";
+    } else
+        return -1;
+
+    ret = metrics_init(NULL);
+    if (ret < 0) {
+        tst_error("Unable to initialize metrics library, %s\n", strerror(errno));
+        free(client);
+        return -1;
+    } else
```

Comment:
Having all the "} else" blocks here is not necessary. A simpler, more readable (IMO) implementation would do:

```
ret = metrics_init(NULL);
if (ret < 0) {
    tst_error("blah");
    free(client);
    return -1;
}

tst_ok("PASS --- TEST: Metrics library initialized\n");
```

Reply:
Done